Loop/Managers/LoopAppManager.swift

```diff
@@ -119,10 +115,26 @@ class LoopAppManager: NSObject {
         }
     }
 
+    private func checkProtectedDataAvailable() {
+        dispatchPrecondition(condition: .onQueue(.main))
+        precondition(state == .checkProtectedDataAvailable)
+
+        guard isProtectedDataAvailable() else {
+            log.default("Protected data not available; deferring launch...")
+            return
```

Comment:
Just trying to understand the state machine here.  If this happens, what calls `launch()` (or `resumeLaunch()`) in this case?

Reply:
`AppDelegate.applicationProtectedDataDidBecomeAvailable` does.